django/api/tests.py

```diff
@@ -77,3 +79,114 @@ def test_restaurants_default_order_by_name(self):
     def test_restaurant_filter_boundingbox(self):
         response = self.c.get("/api/restaurants/?in_bbox=-2,-2,2,2")
         self.assertEqual(response.data["count"], 2)
+
+
+class UserSerializerTestCase(TestCase):
+    def setUp(self):
+        self.test_user = UserFactory()
+        self.test_restaurant_1 = RestaurantFactory.create(
+            __sequence=1, owner=(self.test_user,)
+        )
+        self.test_restaurant_2 = RestaurantFactory.create(
+            __sequence=2, owner=(self.test_user,)
+        )
+
+    def test_user_owns_multiple_restaurants(self):
+        user = User.objects.get(username=self.test_user.username)
+        restaurants = Restaurant.objects.filter(owner=user)
+        self.assertEqual(2, len(restaurants))
+
+    def test_user_is_owner(self):
+        user = User.objects.get(username=self.test_user.username)
+        user_data = UserSerializer(user).data
+        restaurants = Restaurant.objects.filter(owner=user)
+        is_owner = True
+        for restaurant in restaurants:
+            owner = dict(
+                RestaurantSerializer(restaurant).data["properties"].pop("owner")[0]
+            )
+            if owner != user_data:
+                is_owner = False
+        self.assertEqual(True, is_owner)
+
+
+class UserViewsetTestCase(TestCase):
+    def setUp(self):
+        self.c = Client()
+        self.userdata_owner_1 = {
+            "user": {"username": "testOwner1", "password": "testOwner1234"},
+            "restaurant_name": "testOwner1s Pizza Joint",
+            "restaurant_address": "1 Pizza Plaza Philadelphia, PA 12345",
+            "restaurant_loc": {"type": "Point", "coordinates": [125.6, 10.1]},
+        }
+        self.userdata_owner_2 = {
+            "user": {"username": "testOwner2", "password": "testOwner1234"},
+            "restaurant_name": "testOwner2s Pizza Joint",
+            "restaurant_address": "2 Pizza Plaza Philadelphia, PA 12345",
+            "restaurant_loc": {"type": "Point", "coordinates": [125.6, 10.1]},
+        }
+        self.userdata_not_owner = {
+            "user": {"username": "testNotAnOwner1", "password": "testOwner1234"}
+        }
+
+    def test_user_creation_not_owner(self):
+        response = self.c.post(
+            "/api/user/", self.userdata_not_owner, content_type="application/json"
+        )
+        # check created status
+        self.assertEqual(response.status_code, 201)
+        # check user exists
+        does_user_exist = User.objects.filter(
+            username=self.userdata_not_owner["user"]["username"]
+        ).exists()
+        self.assertEqual(True, does_user_exist)
```

Comment:
```suggestion
        self.assertTrue(does_user_exist)
```

(there are a few other places where this suggestion applies as well)

Reply:
You could even do `self.assertTrue(User.objects.filter(...).exists())` if you want.